src/lightning/fabric/fabric.py

```diff
@@ -410,6 +429,7 @@ def backward(self, tensor: Tensor, *args: Any, model: Optional[_FabricModule] =
 
         """
         module = model._forward_module if model is not None else model
+        module, _ = _unwrap_compiled(module)
```

Comment:
Why do you need to unwrap here and in `no_backward_sync`?

Reply:
For completeness, it's optional.